Library/test-helpers/Dockerfile.upstream.c9s

```diff
@@ -1,11 +1,11 @@
 FROM quay.io/centos/centos:stream9
-COPY lime_con_start.sh /usr/local/bin/lime_con_start
 COPY lime_con_install_upstream.sh /usr/local/bin/lime_con_install_upstream
-COPY id_*.pub /root/
 RUN chmod a+x /usr/local/bin/lime_con_* && \
     mkdir -p /mnt/keylime_sources && \
     cp -r /mnt/keylime_sources /var/tmp/keylime_sources && \
-    /usr/local/bin/lime_con_install_upstream && \
-    ssh-keygen -A && mkdir -p /root/.ssh && \
-    cat /root/id_*.pub > /root/.ssh/authorized_keys && \
-    chmod 700 /root/.ssh/authorized_keys
+    /usr/local/bin/lime_con_install_upstream
+
+EXPOSE 9002
+EXPOSE 8892
+
+CMD ["keylime_agent"]
```

Comment:
I added this because I thought this image was used only for the agent. Probably I'll propose completely removing this Dockerfile later, and use the actual upstream Dockerfiles instead (those maintained as part of the upstream repo).

Reply:
Can we use upstream Dockerfiles already or are there any changes still needed?
Also, in order to work with upstream Dockerfiles, maybe we could support storing Dockerfile URL in the `VERIFIER_DOCKERFILE` variable.

Reply:
I'm still fighting a permission issue: I need the keylime user inside the container to have access to the certificate files generated outside the container. 

The problem is that the keylime user inside the container has an arbitrary UID, and I have to change the ownership accordingly. 

I need a way to get the UID of the keylime user inside the container so that I `chown` the files before mounting, or a way to map the keylime user outside the container to the keylime user inside the container.

The only solution I have now is ugly:

- Run the `id -u keylime` command inside the container once to get the UID, `chown` the files, and then start the container.
- Alternatively, instead of `chown` the files to the UID of the keylime user inside the container, I can use `--uidmap` option to map the keylime user outside to the keylime user inside, but I still need the UID of the keylime user inside the container.

It would be great if I could use something like `podman unshare` to do this, but this only exist when running containers as non-root (and we need to run as root to have access to the TPM and event logs). Even better if I could map the UIDs using the username inside the container (something like `--usermap=<host username>:<container username>`).

I believe this is something that affects many services that are executed as root, but later drops privileges to an unprivileged user. Do you have any suggestion?

Reply:
Hmm I don't know if it's possible, but what about run container as non-root and add root capabilities for access to TPM and event log?

Reply:
> Hmm I don't know if it's possible, but what about run container as non-root and add root capabilities for access to TPM and event log?

This is the dream, to be able to run with unprivileged user and still have access to TPM and event log. If possible to do this by adding capabilities, then our work would be much easier.

Reply:
I'll contact the podman guy and ask him about this possibility and let you know.

Reply:
Just for the record: we were informed that capabilites in non-root container are just namespaced capabilities, not real capabilities. This makes it not possible to run the agent in the container as an unprivileged user and provide capabilities to access resources that require higher privileges.